### PR TITLE
Feature/use basic auth order property

### PR DESCRIPTION
**Pull request by floruschbaschan**
Signed-off-by: Florian Ruschbaschan <Florian.Ruschbaschan@bosch-si.com>

### CODE REVIEW

**Comment by schabdo** [comment]
@hawkbit-bot verify please

**Comment by schabdo** [comment]
@hawkbit-bot verify please

**Comment by floruschbaschan** [comment]
Hey, I have checked my profile at https://accounts.eclipse.org and it shows that I have an active Eclipse Contributor Agreement which will expire on 2022-06-05. Also the "Signed-off-by footer" is part of every commit-message. I don't know why the eclipsefdn/eca check fails, everything seems to be fine.

Please advise

**Comment by schabdo** [comment]
@laverman I observed same yesterday in a different PR. The ECA tool seems to be broken. Thanks for getting in touch with the guys from Eclipse regarding this issue

File: hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/WebSecurityConfiguration.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
  * Depending on the ~.auth.enabled property either basic authentication is enabled or access is granted for all requests.
  */
 @EnableWebSecurity
+@Order(SecurityProperties.BASIC_AUTH_ORDER + 1)
```
Comment (schabdo) [inline hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/WebSecurityConfiguration.java]: The property `CONFIGURATION_ENABLED_PROPERTY` is controlling whether or not basic auth is enabled. So I'd make use of it and rather register only one `WebSecurityConfigurerAdapter`(with basic auth or without) than registering always two `WebSecurityConfigurerAdapter` instances and overwrite them by order

Comment (floruschbaschan) [reply hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/WebSecurityConfiguration.java]: Okay, I have changed it. The property "hawkbit.device.simulator.auth.enabled" is now controlling the selected web security configuration.

**Comment by schabdo** [comment]
Please add a licence file for the current year 2020, thanks

**Comment by schabdo** [comment]
Thanks @floruschbaschan 👍

File: hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/http/NoAuthSecurityConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+
+/**
+ * Spring security configuration.
+ * Depending on the ~.auth.enabled property access is granted for all requests.
+ */
+@EnableWebSecurity
```
Comment (bogdan-bondar) [inline hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/http/NoAuthSecurityConfiguration.java]: check if we really need this annotation here, and then probably add it to BasicAuthSecurityConfiguration class

File: hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/http/NoAuthSecurityConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+@ConditionalOnProperty(name = BasicAuthProperties.CONFIGURATION_ENABLED_PROPERTY, havingValue = "false")
```
Comment (bogdan-bondar) [inline hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/http/NoAuthSecurityConfiguration.java]: just to be safe, could you also add matchIfMissing=true?

File: hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/http/BasicAuthSecurityConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        auth.inMemoryAuthentication()
+                .withUser(basicAuthProperties.getUser())
+                .password(passwordEncoder().encode(basicAuthProperties.getPassword()))
+                .roles("ADMIN");
+    }
+
+    @Override
+    protected void configure(final HttpSecurity httpSec) throws Exception {
+        httpSec.csrf().disable().authorizeRequests()
+                .antMatchers("/").permitAll()
```
Comment (bogdan-bondar) [inline hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/http/BasicAuthSecurityConfiguration.java]: I guess the antMatcher is wrong here, shouldn't it be "/**"?

Comment (floruschbaschan) [reply hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/http/BasicAuthSecurityConfiguration.java]: In this case "/", "/**" or "**" should work. But I have changed it to "/**"

**Pull request merged**